src/core/ddsi/include/dds/ddsi/ddsi_serdata.h

```diff
@@ -132,6 +132,11 @@ typedef bool (*ddsi_serdata_eqkey_t) (const struct ddsi_serdata *a, const struct
      returning bufsize-1) if it had to truncate) */
 typedef size_t (*ddsi_serdata_print_t) (const struct ddsi_sertopic *topic, const struct ddsi_serdata *d, char *buf, size_t size);
 
+/* Add keyhash (from serdata) to buffer (forcing md5 when necessary).
+   - key needs to be set within serdata (can already be md5)
+   - buf needs to be at least 16 bytes large */
+typedef void (*ddsi_serdata_add_keyhash_t) (const struct ddsi_serdata *d, char *buf, bool force_md5);
```

Comment:
I'd rather see ``struct nn_keyhash *buf`` here (the name is crappy and it is defined in the wrong place, but as it is used already for ``ddsi_serdata_from_keyhash_t``, using it here without fixing up those definitional matters doesn't introduce any new problems). That type is simply a struct with a single member, an array of 16 bytes.

Secondly, I don't like the name ``add_keyhash``: it doesn't "add" the keyhash as much as it stores it in the provided buffer. In my view the behaviour of function would be more obvious if it were called ``get_keyhash``.